OrbitGl/TimerTrack.cpp

```diff
@@ -0,0 +1,302 @@
+      for (size_t k = 0; k < block.size(); ++k) {
+        TextBox& text_box = block[k];
+        const Timer& timer = text_box.GetTimer();
+        if (min_tick > timer.m_End || max_tick < timer.m_Start) continue;
+        if (timer.m_Start >= min_ignore && timer.m_End <= max_ignore) continue;
+        if (!TimerFilter(timer)) continue;
+
+        UpdateDepth(timer.m_Depth + 1);
+        double start_us = time_graph_->GetUsFromTick(timer.m_Start);
+        double end_us = time_graph_->GetUsFromTick(timer.m_End);
+        double elapsed_us = end_us - start_us;
+        double normalized_start = start_us * inv_time_window;
+        double normalized_length = elapsed_us * inv_time_window;
+        float world_timer_width =
+            static_cast<float>(normalized_length * world_width);
+        float world_timer_x =
+            static_cast<float>(world_start_x + normalized_start * world_width);
+        float world_timer_y = GetYFromDepth(timer.m_Depth);
+
+        bool is_visible_width = normalized_length * canvas->getWidth() > 1;
+        bool is_selected = &text_box == Capture::GSelectedTextBox;
+        bool is_inactive =
+            !Capture::GVisibleFunctionsMap.empty() &&
+            Capture::GVisibleFunctionsMap[timer.m_FunctionAddress] == nullptr;
+
+        Vec2 pos(world_timer_x, world_timer_y);
+        Vec2 size(world_timer_width, box_height_);
+        float z = GlCanvas::Z_VALUE_BOX_ACTIVE;
+        Color color = GetTimerColor(timer, is_selected, is_inactive);
+        text_box.SetPos(pos);
+        text_box.SetSize(size);
+
+        auto user_data = std::make_unique<PickingUserData>(
+            &text_box, [&](PickingID id) { return this->GetBoxTooltip(id); });
+
+        if (is_visible_width) {
+          if (!is_collapsed) {
+            SetTimesliceText(timer, elapsed_us, min_x, &text_box);
+          }
+          batcher->AddShadedBox(pos, size, z, color, PickingID::BOX,
+                                std::move(user_data));
+        } else {
+          auto type = PickingID::LINE;
+          batcher->AddVerticalLine(pos, size[1], z, color, type,
+                                   std::move(user_data));
+          // For lines, we can ignore the entire pixel into which this event
+          // falls. We align this precisely on the pixel x-coordinate of the
+          // current line being drawn (in ticks). If pixel_delta_in_ticks is
+          // zero, we need to avoid dividing by zero, but we also wouldn't
+          // gain anything here.
+          if (pixel_delta_in_ticks != 0) {
+            min_ignore =
+                min_timegraph_tick +
+                ((timer.m_Start - min_timegraph_tick) / pixel_delta_in_ticks) *
+                    pixel_delta_in_ticks;
+            max_ignore = min_ignore + pixel_delta_in_ticks;
+          }
+        }
+      }
+    }
+  }
+}
+
+//-----------------------------------------------------------------------------
+void TimerTrack::OnDrag(int x, int y) { Track::OnDrag(x, y); }
```

Comment:
I'd prefer to omit the method completely if it only calls the parent, so it is clear from the header file already that there is no change in functionality here. What's the reason to have it like this?

Reply:
Yes, fixed.